### PR TITLE
Add 'f' placeholder option for preview pane

### PR DESCRIPTION
**Pull request by srfraser**
If present the contents of the selection will be placed in a temporary file, and the filename will be placed into the string instead.

Addresses comments in #1719

### CODE REVIEW

File: src/terminal.go [hunk 1]
```diff
@@ -1359,8 +1409,8 @@ func (t *Terminal) currentItem() *Item {
 
 func (t *Terminal) buildPlusList(template string, forcePlus bool) (bool, []*Item) {
 	current := t.currentItem()
-	plus, query := hasPreviewFlags(template)
-	if !(query && len(t.input) > 0 || (forcePlus || plus) && len(t.selected) > 0) {
+	plus, query, file := hasPreviewFlags(template)
+	if !(query && len(t.input) > 0 || file || (forcePlus || plus) && len(t.selected) > 0) {
```
[the remark this answers is not in the document]
Comment (srfraser) [reply src/terminal.go]: You're right, it was a holdover in thinking from the stdin patch. Sorry.

File: src/terminal.go [hunk 2]
```diff
@@ -1228,10 +1235,40 @@ func hasPreviewFlags(template string) (plus bool, query bool) {
...
 
+func writeTemporaryFile(data []string) string {
+	f, err := ioutil.TempFile("", "fzf-preview-*")
+	if err != nil {
+		errorExit("Unable to create temporary file")
+	}
+	defer f.Close()
+
+	f.WriteString(strings.Join(data, "\n"))
+	f.WriteString("\n")
```
[the remark this answers is not in the document]
Comment (srfraser) [reply src/terminal.go]: `print0` doesn't currently affect the data passed to the preview window, if I'm reading this correctly. I will add it, and the necessary option. I suspect the easiest way, given the increasing size of `replacePlaceholder`'s arguments is to change it to `func (t *Terminal replacePlaceholder)`

**Comment by junegunn** [comment]
Thanks, let me finish up with a few changes and a test case. Please allow my commits to your branch.

https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by srfraser** [comment]
> Thanks, let me finish up with a few changes and a test case. Please allow my commits to your branch.

Already set! ;) Thanks!

**Comment by junegunn** [comment]
All looks good now. Is there anything else you want to address? I'll update the man page when I release the new version.

**Comment by srfraser** [comment]
It all looks good to me. Thank you for your good feedback, I've learned a lot

**Comment by junegunn** [comment]
Merged, thanks!

**Pull request merged**